src/variety/batten.js

```diff
@@ -0,0 +1,441 @@
+			play: ["shade", "unshade", "completion"]
+		},
+
+		mouseinput_number: function() {
+			if (this.mousestart) {
+				this.inputqnum_excell();
+			}
+		},
+
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart) {
+					this.inputqcmp();
+				}
+				if (this.mousestart || this.mousemove) {
+					this.inputShade();
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart || this.mousemove) {
+					this.inputcheckerboard();
+				}
+				if (this.mouseend && this.notInputted()) {
+					this.inputqnum_excell();
+				}
+			}
+		},
+
+		inputShade: function() {
+			if (this.inputData === 1 || this.inputData === 2) {
+				if (this.board.cellsAreForcedOpposite(this.getcell(), this.mouseCell)) {
+					return;
+				}
+			}
+			this.inputcell();
+		},
+
+		mouseinput_other: function() {
+			if (this.inputMode === "mark-checkerboard") {
+				this.inputcheckerboard();
+			}
+		},
+
+		inputcheckerboard: function() {
+			var cross = this.getcross();
+			var bd = this.board;
+			if (
+				this.prevPos.equals(cross) ||
+				cross.bx < 2 ||
+				cross.bx > bd.maxbx - 2 ||
+				cross.by < 2 ||
+				cross.by > bd.maxby - 2
+			) {
+				return;
+			}
+			this.prevPos.set(cross);
+
+			if (this.inputData === null) {
+				this.inputData = cross.qnum === 1 ? -1 : 1;
+			}
+
+			this.puzzle.opemgr.disCombine = true;
+			cross.setQnum(this.inputData);
+			this.puzzle.opemgr.disCombine = false;
+
+			cross.draw();
+		},
+
+		inputqnum_excell: function() {
+			var excell = this.getcell_excell();
+			if (excell.isnull || excell.group !== "excell") {
+				return;
+			}
+
+			if (excell !== this.cursor.getex()) {
+				this.setcursor(excell);
+			} else {
+				this.inputqnum_main(excell);
+			}
+		},
+
+		inputqcmp: function() {
+			var excell = this.getcell_excell();
+			if (excell.isnull || excell.noNum() || excell.group !== "excell") {
+				return;
+			}
+
+			excell.setQcmp(+!excell.qcmp);
+			excell.draw();
+
+			this.mousereset();
+		}
+	},
+
+	KeyEvent: {
+		enablemake: true,
+		keyinput: function(ca) {
+			if (!this.cursor.getex().isnull) {
+				this.key_inputexcell(ca);
+			} else if (!this.cursor.getx().isnull) {
+				this.key_inputcross(ca);
+			}
+		}
+	},
+
+	TargetCursor: {
+		initCursor: function() {
+			this.init(-1, -1);
+		},
+
+		movedir: function(dir, mv) {
+			var prevx = this.bx;
+			var prevy = this.by;
+
+			this.common.movedir.call(this, dir, mv);
+
+			if (this.isOutsideGrid()) {
+				if (!(this.bx & 1)) {
+					this.bx--;
+					this.by--;
+				}
+			} else {
+				if (this.bx & 1) {
+					this.bx++;
+					if (this.bx > this.maxx) {
+						this.bx -= 2;
+					}
+
+					this.by++;
+					if (this.by > this.maxy) {
+						this.by -= 2;
+					}
+
+					if (this.bx <= 0 || this.by <= 0) {
+						this.bx = prevx;
+						this.by = prevy;
+					}
+				}
+			}
+		},
+
+		isOutsideGrid: function() {
+			return this.bx <= 0 || this.by <= 0;
+		}
+	},
+
+	Cross: {
+		maxnum: 1
```

Comment:
I found a bug related to inputting crosses: You can press 0 when the cursor is inside the grid, and it'll put down a zero clue. This clue is invisible, but it will allow for a checkerboard clue at that location. Duplicating the board will preserve it, while exporting removes it.

Reply:
Apparently the built-in `key_inputcross` doesn't respect `minnum` and there's no way to disable question mark input either so I checked what Mid-loop and No Three do and fixed this analogously.